src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -135,10 +137,13 @@ public void OnStartActivity(Activity activity, object? payload)
                 ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Client);
             }
 
-            // see the spec https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md
-            HttpTagHelper.RequestDataHelper.SetHttpMethodTag(activity, request.Method.Method);
+            if (!IsNet9OrGreater)
+            {
+                // see the spec https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md
+                HttpTagHelper.RequestDataHelper.SetHttpMethodTag(activity, request.Method.Method);
+            }
 
-            if (request.RequestUri != null)
+            if (!IsNet9OrGreater && request.RequestUri != null)
```

Comment:
nit: Seems like this could be moved up into the check on line 140.

```csharp
            if (!IsNet9OrGreater)
            {
                // see the spec https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md
                HttpTagHelper.RequestDataHelper.SetHttpMethodTag(activity, request.Method.Method);

                if (request.RequestUri != null)
                {
                     // ...
                }
            }
```